database/database_test.go

```diff
@@ -234,3 +236,423 @@ func TestStatesWithStatus(t *testing.T) {
+		statusReport := func(format string, args ...interface{}) {
+			formatted := fmt.Sprintf(format, args...)
+			statuses = append(statuses, formatted)
+		}
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+
+		if !reflect.DeepEqual(statuses, tt.wantStatusReports) {
+			t.Errorf("[%s] Incorrect status reports: %#v", tt.description, statuses)
+		}
+	}
+}
+
+// getAndDeleteTests is a struct that is used in testing GetIneligibleDomainStates
+// and DeleteIneligibleDomainStates
+var getAndDeleteTests = []struct {
+	description  string
+	domainNames []string
+	wantStates   []IneligibleDomainState
+}{
+	{
+		"one domain",
+		[]string{"a.test"},
+		[]IneligibleDomainState{
+			{Name: "a.test", Policy: "bulk-1-year", Scans: []Scan{ 
+				{
+					ScanTime: time.Unix(1234,54324),
+					Issues: []hstspreload.Issues{
+						{
+							Errors: []hstspreload.Issue{
+								{
+									Code: "formatting error", 
+									Summary: "missing end of page line",
+									Message: "add a line at the end of the page",
+								},
+							},
+						},
+					},
+				},
+			}},
+		},
+	},
+	{
+		"two domains",
+		[]string{"b.test", "c.test"},
+		[]IneligibleDomainState{{Name: "b.test", Policy: "bulk-18-week", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "formatting error", 
+								Summary: "missing end of page line",
+								Message: "add a line at the end of the page",
+							},
+						},
+					},
+				},
+			},
+		}},
+			{Name: "c.test", Policy: "bulk-18-week", Scans: []Scan{ 
+				{
+					ScanTime: time.Unix(1234,54324),
+					Issues: []hstspreload.Issues{
+						{
+							Errors: []hstspreload.Issue{
+								{
+									Code: "example error", 
+									Summary: "missing example",
+									Message: "add example to code",
+								},
+							},
+						},
+					},
+				},
+			}}},
+	},
+	{
+		"multiple domains",
+		[]string{"e.test", "f.test", "g.test", "h.test", "i.test", "j.test", "k.test"},
+		[]IneligibleDomainState{
+			{Name: "e.test", Policy: "bulk-1-year"},
+			{Name: "f.test", Policy: "bulk-1-year"},
+			{Name: "g.test", Policy: "bulk-18-week"},
+			{Name: "h.test", Policy: "bulk-18-week"},
+			{Name: "i.test", Policy: "bulk-1-year"},
+			{Name: "j.test", Policy: "bulk-18-week"},
+			{Name: "k.test", Policy: "bulk-1-year"},
+		},
+	},
+}
+
+// Test GetIneligibleDomainStates tests getting IneligibleDomainStates from the 
+// database
+func TestGetIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	// test for entry that does not exist
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
```

Comment:
This string looks like it was copied/pasted, as it refers to preloaded domains instead of ineligible domain states.